lib/acts_as_list/active_record/acts/list.rb

```diff
@@ -313,62 +324,26 @@ def decrement_positions_on_lower_items(position=nil)
         def increment_positions_on_all_items
           acts_as_list_list.increment_all
         end
-
-        # Reorders intermediate items to support moving an item from old_position to new_position.
-        def shuffle_positions_on_intermediate_items(old_position, new_position, avoid_id = nil)
-          return if old_position == new_position
-          scope = acts_as_list_list
-
-          if avoid_id
-            scope = scope.where("#{quoted_table_name}.#{self.class.primary_key} != ?", self.class.connection.quote(avoid_id))
-          end
-
-          if old_position < new_position
-            # Decrement position of intermediate items
-            #
-            # e.g., if moving an item from 2 to 5,
-            # move [3, 4, 5] to [2, 3, 4]
-            scope.where(
-              "#{quoted_position_column_with_table_name} > ?", old_position
-            ).where(
-              "#{quoted_position_column_with_table_name} <= ?", new_position
-            ).decrement_all
-          else
-            # Increment position of intermediate items
-            #
-            # e.g., if moving an item from 5 to 2,
-            # move [2, 3, 4] to [3, 4, 5]
-            scope.where(
-              "#{quoted_position_column_with_table_name} >= ?", new_position
-            ).where(
-              "#{quoted_position_column_with_table_name} < ?", old_position
-            ).increment_all
-          end
-        end
-
+        
         def insert_at_position(position)
           return set_list_position(position) if new_record?
           with_lock do
             if in_list?
               old_position = send(position_column).to_i
               return if position == old_position
-              shuffle_positions_on_intermediate_items(old_position, position)
+              # temporary move after bottom with gap, avoiding duplicate values
+              # gap is required to leave room for position increments
+              # positive number will be valid with unique not null check (>= 0) db constraint
+              tmp_position_after_everything = acts_as_list_class.maximum(position_column).to_i + 2
```

Comment:
I think it's handy to assign it to a variable in order to explain what we're doing here. Otherwise it's a but opaque. :)